packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts

```diff
@@ -0,0 +1,398 @@
+    sourceFiles: [
+      assetTransferGo,
+      smartContractGo,
+      goMod,
+      goSum,
+      privateDataCollections,
+    ],
+    collectionsConfigFile: privateDataCollectionName,
+    ccName: contractName,
+    targetOrganizations: [org1Env, org2Env],
+    caFile: `${orgCfgDir}ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem`,
+    ccLabel: "basic-asset-transfer-2",
+    ccLang: ChainCodeProgrammingLanguage.Golang,
+    ccSequence: 1,
+    orderer: "orderer.example.com:7050",
+    ordererTLSHostnameOverride: "orderer.example.com",
+    connTimeout: 60,
+  });
+
+  const { packageIds, lifecycle, success } = res.data;
+  t.equal(res.status, 200, "res.status === 200 OK");
+  t.true(success, "res.data.success === true");
+
+  const {
+    approveForMyOrgList,
+    installList,
+    queryInstalledList,
+    commit,
+    packaging,
+    queryCommitted,
+  } = lifecycle;
+
+  Checks.truthy(packageIds, `packageIds truthy OK`);
+  Checks.truthy(
+    Array.isArray(packageIds),
+    `Array.isArray(packageIds) truthy OK`,
+  );
+  Checks.truthy(approveForMyOrgList, `approveForMyOrgList truthy OK`);
+  Checks.truthy(
+    Array.isArray(approveForMyOrgList),
+    `Array.isArray(approveForMyOrgList) truthy OK`,
+  );
+  Checks.truthy(installList, `installList truthy OK`);
+  Checks.truthy(
+    Array.isArray(installList),
+    `Array.isArray(installList) truthy OK`,
+  );
+  Checks.truthy(queryInstalledList, `queryInstalledList truthy OK`);
+  Checks.truthy(
+    Array.isArray(queryInstalledList),
+    `Array.isArray(queryInstalledList) truthy OK`,
+  );
+  Checks.truthy(commit, `commit truthy OK`);
+  // Checks.truthy(init, `init truthy OK`);
+  Checks.truthy(packaging, `packaging truthy OK`);
+  Checks.truthy(queryCommitted, `queryCommitted truthy OK`);
+
+  // FIXME - without this wait it randomly fails with an error claiming that
+  // the endorsment was impossible to be obtained. The fabric-samples script
+  // does the same thing, it just waits 10 seconds for good measure so there
+  // might not be a way for us to avoid doing this, but if there is a way we
+  // absolutely should not have timeouts like this, anywhere...
+  await new Promise((resolve) => setTimeout(resolve, 10000));
+
+  const assetId = uuidv4();
+  const assetType = "asset";
+
+  const assetData = {
+    objectType: assetType,
+    assetID: assetId,
+    color: "gray",
+    size: 3,
+    appraisedValue: 500,
+  };
+
+  //Chaincode-specific method requires attribute asset_properties
+  const rawTmpData = {
+    asset_properties: assetData,
+  };
+
+  // CreateAsset(id string, color string, size int, owner string, appraisedValue int)
+  const createRes = await apiClient.runTransactionV1({
+    transientData: rawTmpData,
+    contractName,
+    channelName,
+    //objectType, assetID, color, size, appraisedvalue
+    params: [],
+    methodName: "CreateAsset",
+    invocationType: FabricContractInvocationType.SENDPRIVATE,
```

Comment:
@RafaelAPB Is there something in the output of the call that we could further use to validate that the whole thing works or not? Or does it not crashing guarantees that everything related to the private data (including the initial collection config json) was working as expected? Sorry if this is a beginner question I still haven't had time to read up on how exactly Fabric does this so I'm just checking.

Reply:
Good point - I believe the tx receipt is the same. The test cases show that this feature is indeed working (but can be expanded)

Reply:
@RafaelAPB Understood, thank you for the explanation!